pkg/querier/worker/frontend_processor_test.go

```diff
@@ -0,0 +1,69 @@
+package worker
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/atomic"
+	"google.golang.org/grpc"
+
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/cortexproject/cortex/pkg/util/test"
+)
+
+func TestRecvFailDoesntCancelProcess(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// We use random port here, hopefully without any gRPC server.
+	cc, err := grpc.DialContext(ctx, "localhost:999", grpc.WithInsecure())
+	require.NoError(t, err)
+
+	cfg := Config{}
+	mgr := newFrontendProcessor(cfg, nil, util.Logger)
+	running := atomic.NewBool(false)
+	go func() {
+		running.Store(true)
+		defer running.Store(false)
+
+		mgr.processQueriesOnSingleStream(ctx, cc, "test:12345")
+	}()
+
+	time.Sleep(50 * time.Millisecond)
+	assert.Equal(t, true, running.Load())
+
+	cancel()
+	test.Poll(t, 100*time.Millisecond, false, func() interface{} {
```

Comment:
The 100ms polling timeouts in these tests may trigger false positives on slow machines. I would suggest increasing it to 1s.

_Same comment applies to `TestServeCancelStopsProcess`._